refactor(formatter): simplify and correct member expression checks

Dunqing opened the pull request:
Follow-up after https://github.com/oxc-project/oxc/pull/11767.

I've checked them one by one, corresponding to the Biome's implementation. It still has a regression test, but it is not exactly related to the `AstKind::MemberExpression` change. I will fix that in another PR, but not now.

Dunqing commented:
* **#11904** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11904?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11904?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=11904).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camchenry reviewed (approved):
Sorry about all of the noise here. Thanks for cleaning this up.

Dunqing committed:
refactor(formatter): simplify and correct member expression checks (#11904)

Follow-up after https://github.com/oxc-project/oxc/pull/11767.

I've checked them one by one, corresponding to the Biome's implementation. It still has a regression test, but it is not exactly related to the `AstKind::MemberExpression` change. I will fix that in another PR, but not now.
